Doc/library/shelve.rst

```diff
@@ -41,13 +42,25 @@ lots of shared  sub-objects.  The keys are ordinary strings.
    determine which accessed entries are mutable, nor which ones were actually
    mutated).
 
+   By default, :mod:`shelve` uses :func:`pickle.dumps` and :func:`pickle.loads`
+   for serializing and deserializing. This can be changed by supplying
+   *serializer* and *deserializer*, respectively. The *serializer* argument
+   should be a function that takes an object and returns its representation
```

Comment:
Mention that the function must also take a protocol argument that may be ignored, namely, it will be called as

```py
data = serializer(obj, protocol)
```

Reply:
> 
The *serializer* argument should be a function that takes an object and  the *protocol* argument passed to the open function and returns its representation as a :term:`bytes-like object`; *protocol* argument that may be ignored by the function.

But it might be a bit confusing, because the `serializer` function [uses](https://github.com/python/cpython/pull/118065/files#diff-3a8b6356369487dc852c6841e6710e162a5e740b8fefdbcfc3df4fe834e756cbR137) the `protocol` argument passed to the open function.

What do you think about this? @picnixz

Reply:
I think your suggestion is fine because the serializer may depend on the protocol or not and yes, it should be like this IMO. I mean, maybe your serializer does something different depending on the protocol itself, like patching some parts. Therefore, I don't think it would be "confusing", but if you want to make it less confusing, just mention how the serializer is being called (a good example is worth a thousand words).